java/src/com/ibm/streamsx/topology/Topology.java

```diff
@@ -502,6 +502,34 @@ public void addJarDependency(String location) {
     public void addClassDependency(Class<?> clazz) {
         JavaFunctional.addClassDependency(this, clazz); 
     }
+
+    /**
+     * Add file or directory tree {@code location} to directory
+     * {@code dstDirName} in the application bundle, making it
+     * available to the application at runtime.
+     * <p>
+     * Use {@link #addClassDependency(Class)} or {@link #addJarDependency(String)}
+     * to add class or jar dependencies.
+     * <p>
+     * Legal values for {@code dstDirName} are {@code etc} or {@code opt}.
+     * <p>
+     * e.g.,
+     * <pre>
+     * // add "myConfigFile" to the bundle's "etc" directory
+     * addFileDependency("etc", "/tmp/myConfigFile");
+     * 
+     * // add "myApp" directory tree to the bundle's "etc" directory
+     * addFileDependency("etc", "/tmp/myApp");
+     * </pre>
+     * 
+     * @param dstDirName name of directory in the bundle
+     * @param location path to a file or directory
+     * @throws IllegalArgumentException if {@code dstDirName} is not {@code etc}
+     *     or {@code opt}, or {@code location} is not a file or directory.
+     */
+    public void addFileDependency(String dstDirName, String location) {
```

Comment:
Isn't it typical to have the order be source then destination, e.g. linux command line copy, Java Files.copy etc.

Need to state that these are (for SPL apps) etc/opt in the application directory.